### PR TITLE
fix: update cucumber tests for walletffi.feature

### PR DESCRIPTION
**Pull request by StriderDM**
Description
Made types more explicit for ffi interface and easier to maintain going forwards.
Initialized InterfaceFFI in world rather than WalletFFIClient.
Updated steps in WalletFFI.feature
Updated some const variables in wrapper classes to be let.
Better handling of callback function pointer variables in wallet.
Added additional steps to tests.
Downgrade code to be compatible with nodejs v12.

Motivation and Context

How Has This Been Tested?
nvm use v12.22.6 && ./node_modules/.bin/cucumber-js features/WalletFFI.feature

Run log:
[run.log](https://github.com/tari-project/tari/files/7097096/run.log)

NodeJS version downgrade:
https://github.com/tari-labs/docker/pull/16

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -129,7 +129,7 @@ commands:
           when: always
       - run:
           name: Run ffi cucumber scenarios
-          command: cd integration_tests && mkdir -p cucumber_output && node_modules/.bin/cucumber-js --tags "not @long-running and not @broken and not @flaky and @wallet-ffi" --format json:cucumber_output/tests_ffi.cucumber --exit
+          command: cd integration_tests && mkdir -p cucumber_output && touch cucumber_output/tests_ffi.cucumber && node_modules/.bin/cucumber-js --tags "not @long-running and not @broken and not @flaky and @wallet-ffi" --format json:cucumber_output/tests_ffi.cucumber --exit
```
Comment (stringhandler) [inline .circleci/config.yml]:
```suggestion
          command: cd integration_tests && mkdir -p cucumber_output && touch cucumber_output/tests_ffi.cucumber && node_modules/.bin/cucumber-js --tags "not @long-running and not @broken and not @flaky and @wallet-ffi" --format json:cucumber_output/tests_ffi.cucumber --exit
          when: always
```

File: integration_tests/features/support/steps.js [hunk 2]
```diff
@@ -3826,15 +3826,15 @@ Then(
...
-      },
-      true,
-      700 * 1000,
-      5 * 1000,
-      5
-    );
+    let count = 0;
+
+    while (!(wallet.getCounters().received >= amount)) {
+      await sleep(1000);
```
Comment (stringhandler) [inline integration_tests/features/support/steps.js]: Why is this better?

File: integration_tests/features/support/steps.js [hunk 3]
```diff
@@ -3860,15 +3860,15 @@ Then(
...
-      },
-      true,
-      700 * 1000,
-      5 * 1000,
-      5
-    );
+    let count = 0;
+
+    while (!(wallet.getCounters().finalized >= amount)) {
+      await sleep(1000);
```
Comment (stringhandler) [inline integration_tests/features/support/steps.js]: Same question

**Review by stringhandler — COMMENTED** [review]
What's wrong with the waitFor method that it needs to be replaced. Wouldn't it be better to fix those?

File: .circleci/config.yml (hunk 1 above)
Comment (stringhandler) [reply .circleci/config.yml]: This runs the FFI test even if the previous step failed

Comment (StriderDM) [reply .circleci/config.yml]: Thanks.

File: integration_tests/features/support/steps.js (hunk 2 above)
Comment (StriderDM) [reply integration_tests/features/support/steps.js]: Wrote the one like that and updated the others so they were all all running the same code, that's the only reason.

Would you like me to change it back to waitFor for all of them?

Comment (stringhandler) [reply integration_tests/features/support/steps.js]: yes, unless there is a reason to not use waitFor?

**Review by stringhandler — DISMISSED** [review]
utACK

**Review by stringhandler — APPROVED** [review]

**Pull request merged**